### ISSUE TITLE
Missing E1120 for the incorrectly called class methods of objects passed to the functions 

### ISSUE THREAD

**Issue by manefix** [PyCQA/pylint#5116]
### Bug description

1. create class with some method with any mandatory field, e.g. 'error_message' 
2. create some function with one argument accepted with type of that class created above
3. inside that function just call the class method of the passed object with no arguments
4. Run pylint vs the code above

**Expected Result**

E1120: No value for argument 'error_message' in method call (no-value-for-parameter)

**Actual Result**

Nothing is reported, and this is pretty common situation

Sample that is required to work but it is not: 

```python
class Report:
    def add_error(self, error_message: str):
        print(self, error_message)


def main(report: Report):
    report.add_error() # here we should pass printable value, but
    #  forgot about it, pycharm will remind us, but not pylint... 

```

Currently working: 

```python
class Report:
    def add_error(self, error_message: str):
        print(self, error_message)

report = Report()
report.add_error() # outside the function this code will cause pylint violations
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
empty
```

### Expected behavior

E1120: No value for argument 'error_message' in method call (no-value-for-parameter)

### Pylint version

```shell
pylint 2.11.1
```

### OS / Environment

MacOS BigSur 11.6

**Comment by Pierre-Sassoulas** [comment]
Closing as duplicate of #4813
